src/redux/reducers/planReducer.js

```diff
@@ -0,0 +1,148 @@
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.ADD_PLAN_TO_DB_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.ADD_PLAN_TO_STATE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced,
+        plans: updateMultiDimensionalArray(
+          state.plans,
+          action.planToAdd.trip_id,
+          action.planToAdd.e_id,
+          action.planToAdd
+        )
+      };
+    case PlanActions.UPDATE_PLAN_IN_DB_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.UPDATE_PLAN_IN_DB_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.UPDATE_PLAN_IN_DB_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_IN_DB_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_IN_DB_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_IN_DB_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_IN_STATE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced,
+        plans: filterMultiDimensionalArray(
+          state.plans,
+          action.planToDelete.trip_id,
+          action.planToDelete.e_id
+        )
+      };
+    case PlanActions.GET_PLAN_INFO_FROM_DB_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.GET_PLAN_INFO_FROM_DB_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.GET_PLAN_INFO_FROM_DB_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
```

Comment:
I'm wondering if this should affect the plans array in the state. In the TripReducer, there's a similar method that does update upon success getting from the db.

Reply:
I don't completely understand this point. Which action are you exactly commenting on?

Reply:
I suspect that this one doesn’t matter. I just had a consistency issue, and made a comment in tripReducer line 34 to say that a get from database action should either:

a) NOT update the items locally (as is the case in this example), or
b) DO update the items locally (as in the case in the trips)

By convention, I’d advocate we choose one way of doing things or the other

Reply:
Never mind! Saw your comment elsewhere. You can ignore this.